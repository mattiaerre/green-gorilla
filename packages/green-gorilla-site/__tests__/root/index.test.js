const fetch = require('node-fetch');
const index = require('../../src/root');

describe('index', () => {
  const config = { appid: 'appid' };
  const root = index(config);

  test('root API', () => {
    expect(root).toMatchSnapshot();
    const FUNCTION = 'function';
    expect(typeof root.currentWeather).toBe(FUNCTION);
    expect(typeof root.info).toBe(FUNCTION);
  });

  test('currentWeather()', async () => {
    const weather = {};
    fetch.mockResponse(JSON.stringify(weather));

    const currentWeather = await root.currentWeather({ city: 'London' });

    expect(currentWeather).toEqual(weather);
    expect(currentWeather).toMatchSnapshot();
  });

  test('fetch', () => {
    expect(fetch).toHaveBeenCalled();
    expect(fetch.mock.calls).toMatchSnapshot();
  });

  test('info()', () => {
    expect(root.info()).toMatchSnapshot();
  });
});
