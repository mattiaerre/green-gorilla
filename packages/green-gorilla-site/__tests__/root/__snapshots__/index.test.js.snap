// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index currentWeather() 1`] = `Object {}`;

exports[`index fetch 1`] = `
Array [
  Array [
    "http://api.openweathermap.org/data/2.5/weather?q=London&appid=appid&units=metric",
  ],
]
`;

exports[`index info() 1`] = `
Object {
  "name": "âžœ ~",
  "version": "1.4.0",
}
`;

exports[`index root 1`] = `
Object {
  "currentWeather": [Function],
  "info": [Function],
}
`;
